client/src/transmitter/index.ts

```diff
@@ -1,33 +1,75 @@
-import isIp from 'is-ip'
-
 import { copyToClipboard } from '../clipboard'
 import cast from './cast'
+import isLocalOrIp from './util/isLocalOrIp'
+
+const { body } = document
+
+const _video = document.getElementById('video') as HTMLVideoElement
 
-const isLocalHost = (hostname) => {
-  return hostname === 'localhost' || hostname.endsWith('.localhost')
+const _canvas = document.getElementById('canvas') as HTMLCanvasElement
+
+function resizeCanvas() {
+  const { innerWidth, innerHeight } = window
+
+  _canvas.width = innerWidth
+  _canvas.height = innerHeight
 }
 
-const { body } = document
+window.addEventListener('resize', () => {
+  resizeCanvas()
+})
 
-const video = document.getElementById('video')
-const playbackUrl = document.getElementById('playbackUrl')
+resizeCanvas()
 
-const record_container = document.getElementById('record-container')
-const record = document.getElementById('record')
+// @ts-ignore
+const _canvas_stream = _canvas.captureStream()
 
-const media_container = document.getElementById('media-container')
-const media = document.getElementById('media')
+const _canvas_ctx = _canvas.getContext('2d')
 
-let media_display = false
+let _stream: MediaStream
 
-let media_display_returned = false
+const _playback_url = document.getElementById('playback-url')
+_playback_url.onclick = () => {
+  copyToClipboard(_playback_url.innerText)
+}
 
-playbackUrl.onclick = () => {
-  copyToClipboard(playbackUrl.innerText)
+const _record_container = document.getElementById('record-container')
+_record_container.style.display = 'block'
+_record_container.onclick = () => {
+  if (_curr_cast) {
+    stopRecording()
+  } else {
+    if (_stream) {
+      startRecording(_stream)
+    }
+  }
```

Comment:
**style:** 
```suggestion
  } else if (_stream) {
    startRecording(_stream)
  }
```

Reply:
Style and Verbosity should not duel.

Reply:
Readability always wins

Reply:
Readability and Verbosity do not duel.

Reply:
I'm still trying to register that "verbosity" concept as a good thing xD

Reply:
There's no Readability rule that says "less is more readable". 

I would say both versions are pretty much equally readable, although one is more verbose than the other as in "it takes more structure and space to express the same concept". 

One point in favor of more Verbosity is that it makes Complexity (like conditionals) more explicit and they are usually better to debug (if ever) and refactor (if ever).

Reply:
Yeah we talked about this the other day. I think the "taking more structure and space to express the same concept" is not only in the text file, but also in our brains though, which have to make multiple interpretation hops to read the code. Like here I read the `else,` then I read the `if X`, while with the language construct that our coder brains have gotten pretty much used to it'd be just an "elif X" and "next line of code". The multiple nesting also affects some reading performance IMO.

Regarding explicitness, I think this is one of the cases where it does more harm than good, especially because of that asymmetry, where a code will be written 1x, debugged 10x and read 100x. 

So, in my own framework, optimizing for debuggability shouldn't compromise readability, gotta find the sweet spot. Maybe here the `if` above could `return;`, then we don't need the `else` and nesting here?

Reply:
This is a hardcore one. 

Something smaller is going to influence Reading Perfomance in a very obvious way. 

At the same time, every reduction fundamentally hides Complexity. On the surface, it would really look like the code is more Readable (it is smaller after all), but I think, as a rule of thumb, what's happening is that we are actually transferring Code Translation work to the brain, because it will necessarily unwrap the reductions to build the tree model that would otherwise be already explicit in the code. In this model, Debugability, Verbosity, Readability and Understanding are all aligned; there's never doubt.

That said, I don't think this is really a big deal. There's a chance you are objectively right about this case, but I'd say the deviation is negligible.